src/Extensions.Hosting.AsyncInitialization/DependencyInjection/AsyncInitializationServiceCollectionExtensions.cs

```diff
@@ -95,28 +95,31 @@ public static IServiceCollection AddAsyncInitializer(this IServiceCollection ser
         /// <param name="services">The <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceCollection" /> to add the service to.</param>
         /// <param name="initializer">The delegate that performs async initialization.</param>
         /// <returns>A reference to this instance after the operation has completed.</returns>
-        public static IServiceCollection AddAsyncInitializer(this IServiceCollection services, Func<Task> initializer)
+        public static IServiceCollection AddAsyncInitializer(this IServiceCollection services, Func<CancellationToken, Task> initializer)
```

Comment:
It would be better to keep the existing method and add a new overload with the cancellation token. This way, people who are already using this method won't need to change their code when they upgrade.

The old method can call the new one, passing `_ => initializer()` for the delegate.